src/overlays/actors/ovl_En_Sc_Ruppe/z_en_sc_ruppe.c

```diff
@@ -31,29 +44,157 @@ const ActorInit En_Sc_Ruppe_InitVars = {
+                return true;
+            }
+            break;
+        case RUPPE_ORANGE:
+            if (gSaveContext.save.weekEventReg[54] & 2) {
+                gSaveContext.save.weekEventReg[54] &= (u8)~2;
+                return true;
+            }
+            break;
+        case RUPPE_PURPLE:
+            if (gSaveContext.save.weekEventReg[54] & 4) {
+                gSaveContext.save.weekEventReg[54] &= (u8)~4;
+                return true;
+            }
+            break;
+        case RUPPE_SILVER:
+            if ((gSaveContext.save.weekEventReg[54] & 8)) {
+                gSaveContext.save.weekEventReg[54] &= (u8)~8;
+                return true;
+            }
+            break;
+    }
+    return false;
+}
+
+void func_80BD6A8C(EnScRuppe* this, GlobalContext* globalCtx) {
+    if (this->collider.base.ocFlags1 & OC1_HIT) {
+        this->ruppeDisplayTime = 0;
+        this->actor.gravity = 0.0f;
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_SY_GET_RUPY);
+        func_801159EC(sRupeeInfo[this->ruppeIndex].amount);
+        this->actionFunc = func_80BD6B18;
+    }
+    this->actor.shape.rot.y += 0x1F4;
+    Actor_MoveWithGravity(&this->actor);
+}
+
+void func_80BD6B18(EnScRuppe* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (this->ruppeDisplayTime > 30) {
+        if (func_80BD697C(this->ruppeIndex)) {
+            Actor_MarkForDeath(&this->actor);
+        }
+    } else {
+        f32 scale;
+
+        this->ruppeDisplayTime++;
+        this->actor.world.pos = player->actor.world.pos;
+        this->actor.world.pos.y += 40.0f;
+        scale = (30.0f - this->ruppeDisplayTime) * 0.001f;
+        Actor_SetScale(&this->actor, scale);
+    }
+    this->actor.shape.rot.y += 0x3E8;
+    Actor_MoveWithGravity(&this->actor);
+}
+
+void EnScRuppe_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnScRuppe* this = THIS;
+    ColliderCylinder* collider = &this->collider;
+
+    Collider_InitCylinder(globalCtx, collider);
+    Collider_InitAndSetCylinder(globalCtx, collider, &this->actor, &sCylinderInit);
+    Actor_SetScale(&this->actor, 0.03f);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 10.0f);
+    this->ruppeIndex = SCRUPPE_GET_PARAMS(thisx);
+    if ((this->ruppeIndex < RUPPE_GREEN) || (this->ruppeIndex >= RUPPE_SILVER)) {
+        this->ruppeIndex = RUPPE_GREEN;
+    }
+    this->actor.speedXZ = 0.0f;
+    this->actionFunc = func_80BD6A8C;
+    this->actor.gravity = -0.5f;
+    this->actor.shape.yOffset = 700.0f;
+}
+
+void EnScRuppe_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnScRuppe* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnScRuppe_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnScRuppe* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+    EnScRuppe_UpdateCollision(this, globalCtx);
+}
+
+void EnScRuppe_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    Gfx* opa;
```

Comment:
Looks like another temp that's just padding, so
```suggestion
    s32 pad;
```